### PR TITLE
 usb: dfu: set bwPollTimeout dynamically

### LINKED ISSUE #8734: USB: DFU timeout on nRF52840 due to long flash erase
Attempt of downloading new firmware with dfu-util on nRF52840 results in timeout due to flash erase operation taking too much time (flash_img_buffered_write() called from dfu_flash_write() in usb_dfu.c.

This problem could be solved by fragmenting whole flash erase operation into smaller page erases when needed.

Working USB driver for testing is available on this fork/branch:  https://github.com/pawelzadrozniak/zephyr/tree/nrfx_usbd3 .

### PR DESCRIPTION
**Pull request by jfischer-no**
Set bwPollTimeout for DFU_GETSTATUS request dynamically.
For now, adjust bwPollTimeout only during DNLOAD stage.

Fixes #8734

### CODE REVIEW

**Comment by jfischer-phytec-iot** [comment]
I tried to touch as little as possible. Side effects possible.

@Qbicz please test (works for me on reel board)
@tormodvolden FYI

File: subsys/usb/class/Kconfig [hunk 1]
```diff
@@ -152,3 +152,19 @@ config USB_DFU_DETACH_TIMEOUT
...
+config USB_DFU_DEFAULT_POLLTIMEOUT
+	int
+	depends on USB_DFU_CLASS
+	default 256
+	range 0 1000
+	help
+	  Default value for bwPollTimeout (in ms)
+
+config USB_DFU_DNLOAD_POLLTIMEOUT
+	int
```
Comment (Qbicz) [inline subsys/usb/class/Kconfig]: Can we add a prompt so it will be user-configurable in case this is needed to be changed?

**Review by Qbicz — APPROVED** [review]
@jfischer-phytec-iot thanks. I tested this and I was able to perform DFU with standard dfu-util which has 5 second USB operation timeout on a nrf52840_pca10056 which has longer erase time.

Thanks @tormodvolden for spotting this. I will close the issue in dfu-util once this is merged.

File: subsys/usb/class/Kconfig (hunk 1 above)
Comment (jfischer-phytec-iot) [reply subsys/usb/class/Kconfig]: sorry, the prompt should be there, will fix

Comment (jfischer-phytec-iot) [reply subsys/usb/class/Kconfig]: @Qbicz fixed

Comment (Qbicz) [reply subsys/usb/class/Kconfig]: Great. For me PR is fine

**Review by Qbicz — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Comment by carlescufi** [comment]
@barsok can you test this today?

**Review by carlescufi — APPROVED** [review]

**Comment by Qbicz** [comment]
@jfischer-phytec-iot what's the preferred way to use 2 .conf files together? We tried with @barsok and the only way that worked was to create a CMake list with 
`cmake -GNinja -DBOARD=$B -DCONF_FILE="overlay-composite-cdc-dfu.conf;prj.conf" .. `

**Comment by jfischer-phytec-iot** [comment]
@Qbicz I do it like this, too.

**Comment by barsok** [comment]
@carlescufi I have tested the code and I have found presumably unwanted behaviour:    [reset] -> dfu read, bwPollTimeout==256 -> dfu download -> error during download,  bwPollTimeout remains at 10000, -> dfu upload   bwPollTimeout==10000.   In English, an error during download does not bring bwPollTimeout to default value

**Comment by carlescufi** [comment]
@jfischer-phytec-iot can you confirm the report from @barsok?

**Pull request merged**